lib/src/main/java/io/ably/lib/types/ChannelOptions.java

```diff
@@ -25,33 +22,40 @@
      * Whether or not this ChannelOptions is encrypted.
      */
     public boolean encrypted;
-    
+
     public boolean hasModes() {
         return null != modes && 0 != modes.length;
     }
-    
+
     public boolean hasParams() {
         return null != params && !params.isEmpty();
     }
-    
+
     public int getModeFlags() {
         int flags = 0;
         for (final ChannelMode mode : modes) {
             flags |= mode.getMask();
         }
         return flags;
     }
-    
-    public ChannelCipher getCipher() throws AblyException {
```

Comment:
Since this is a public-facing method in the API that we're removing, it can potentially cause problems for some SDK users that are using this `getCipher()` method. Therefore, maybe we should deprecate it somehow instead of removing it? Or make it backwards compatible? WDYT? :wink:

Reply:
It was annotated as internal, but I respect your point and I think my approach has perhaps been a bit heavy handed. I'm taking a look at refactoring now. Shall move PR back to Draft state to reflect this.

Reply:
Sorry for the noise, I initially ran at this in https://github.com/ably/ably-java/pull/746/commits/546b71337a47328c84652005a83dbc3cf7842f46 and then realised that I had completely missed the point, so I then fixed it up in https://github.com/ably/ably-java/pull/746/commits/339178346baf14c5134fd5d94ea38b332738efff.

Reply:
Looks good, thanks :smile: